ui/hooks/metamask-notifications/useProfileSyncing.ts

```diff
@@ -112,3 +118,63 @@ export function useSetIsProfileSyncingEnabled(): {
 
   return { setIsProfileSyncingEnabled, error };
 }
+
+/**
+ * Custom hook to dispatch account syncing.
+ *
+ * @returns An object containing the `dispatchAccountSyncing` function, boolean `shouldDispatchAccountSyncing`,
+ * loading state, and error state.
+ */
+export const useAccountSyncing = () => {
+  const dispatch = useDispatch();
+
+  const [isLoading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<unknown>(null);
+
+  const isProfileSyncingEnabled = useSelector(selectIsProfileSyncingEnabled);
+  const basicFunctionality = useSelector(getUseExternalServices);
+  const isUnlocked = useSelector(getIsUnlocked);
+  const isSignedIn = useSelector(selectIsSignedIn);
+
+  const shouldDispatchAccountSyncing = useMemo(
+    () =>
+      basicFunctionality && isProfileSyncingEnabled && isUnlocked && isSignedIn,
+    [basicFunctionality, isProfileSyncingEnabled, isUnlocked, isSignedIn],
+  );
+
+  const dispatchAccountSyncing = useCallback(async () => {
+    setLoading(true);
+    setError(null);
+
+    try {
+      if (!shouldDispatchAccountSyncing) {
+        return;
+      }
+      await dispatch(syncInternalAccountsWithUserStorage());
+    } catch (e) {
+      log.error(e);
+      setError(e instanceof Error ? e.message : 'An unexpected error occurred');
+      throw e;
```

Comment:
Q - We do this with our other hooks. I'm not entirely convinced, should hooks throw?